pkg/aws/eni/instances.go

```diff
@@ -278,6 +279,51 @@ func (m *InstancesManager) UpdateENI(instanceID string, eni *eniTypes.ENI) {
 	m.mutex.Unlock()
 }
 
+func (m *InstancesManager) AddIPsToENI(instanceID string, eniID string, ips []string) {
+	m.modifyIPsToENI(instanceID, eniID, ips, true)
+}
+
+func (m *InstancesManager) RemoveIPsFromENI(instanceID string, eniID string, ips []string) {
+	m.modifyIPsToENI(instanceID, eniID, ips, false)
+}
+
+func (m *InstancesManager) modifyIPsToENI(instanceID string, eniID string, ips []string, isAdd bool) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	ifaces, ok := m.instances.GetInterface(instanceID, eniID)
+	if !ok {
+		log.WithFields(logrus.Fields{
+			"instance": instanceID,
+			"eni":      eniID,
+		}).Warning("ENI not found")
+		return
+	}
+
+	eniIntf := ifaces.Resource.DeepCopyInterface()
+	eni, ok := eniIntf.(*eniTypes.ENI)
+	if !ok {
+		log.WithFields(logrus.Fields{
+			"instance": instanceID,
+			"eni":      eniID,
+		}).Warning("Unexpected resource type, expected *eniTypes.ENI")
+		return
+	}
+	if isAdd {
+		for _, ip := range ips {
+			if !slices.Contains(eni.Addresses, ip) {
+				eni.Addresses = append(eni.Addresses, ip)
+			}
+		}
+	} else {
+		for _, ip := range ips {
+			if idx := slices.Index(eni.Addresses, ip); idx != -1 {
+				eni.Addresses = slices.Delete(eni.Addresses, idx, idx+1)
+			}
```

Comment:
nit: You should be able to use `slices.DeleteFunc` instead of `Index+Delete`, it would simplify the code slightly